x-pack/plugins/ml/public/components/form_label/form_label.js

```diff
@@ -4,34 +4,31 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import './styles/main.less';
 
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
 
-import { uiModules } from 'ui/modules';
-const module = uiModules.get('apps/ml');
+import { JsonTooltip } from '../json_tooltip/json_tooltip';
 
-// directive for creating a form label including a hoverable icon
-// to provide additional information in a tooltip. label and tooltip
-// text elements get unique ids based on label-id so they can be
-// referenced by attributes, for example:
-//
-// <ml-form-label label-id="uid">Label Text</ml-form-label>
-// <input
-//   type="text"
-//   aria-labelledby="ml_aria_label_uid"
-//   aria-describedby="ml_aria_description_uid"
-// />
-module.directive('mlFormLabel', function () {
-  return {
-    scope: {
-      labelId: '@',
-      tooltipAppendToBody: '@'
-    },
-    restrict: 'E',
-    replace: false,
-    transclude: true,
-    template: `
-      <label class="kuiFormLabel" id="ml_aria_label_{{labelId}}" ng-transclude></label>
-      <i ml-info-icon="{{labelId}}" position="top" />
-    `
-  };
-});
+// Writing this as a class based component because stateless components
+// cannot use ref(). Once angular is completely gone this can be rewritten
+// as a function stateless component.
+export class FormLabel extends Component {
+  constructor(props) {
+    super(props);
+    this.labelRef = React.createRef();
+  }
+  render() {
+    const { labelId, children } = this.props;
+    return (
+      <React.Fragment>
+        <label className="kuiFormLabel" id={`ml_aria_label_${labelId}`} ref={this.labelRef}>{children}</label>
```

Comment:
Should this still be using the `kuiFormLabel` class? Is there an eui style we can use instead? If you do change the `kuiFormLabel` class, the Jest file will need updating too.

Reply:
In 0bb4939 I updated the component so it defaults to `euiFormLabel` but you can override it with another class, for example with `kuiFormLabel` when used within an angular context (necessary so we don't end up with mixed styles when a form uses both `ml-form-label` and plain labels using `kuiFormLabel`).